components/rollup_node/README.md

```diff
@@ -0,0 +1,70 @@
+| L1 Verifier           | X                | X                    |                      |                |                 |
+| Replicating Verifier  | X                | X                    | X                    |                |                 |
+| Sequencer             | X                | X                    |                      | X              | X               |
+
+## Components
+
+### [Execution Engine][exec-engine]
+
+The execution engine implements the [execution specification][execution-spec].  Most L1 clients will soon be converted into execution engines. The rollup client will communicate to the engine via a bidirectional JSON-RPC interface ([WIP][execution-engine-rpc]).
+
+One of the main goal of the rollup client is to use the execution engine without modification.
+
+All rollup nodes run an EE.
+
+### [Consensus Layer][consensus-layer]
+
+The consensus layer connects to an L1 node, and tracks the feed data comprising of deposits and sequencer batches. Based on the rules of [block generation][block-gen], it is able to compute sets ("epochs") of L2 blocks as a stateless, pure function of ranges of L1 blocks.
+
+Together with the Execution Engine, it uses the Engine API to progress the L2 state as new data comes in from L1. The consensus layer will reorganize the L2 state if it detects a between the EE and what's on L1, and provides a way for other services to subscribe to reorgs.
+
+All rollup nodes run a CL.
+
+### [Sequencer Replicator][sequencer-replicator]
+
+The sequencer replicator connects directly to the sequencer. It requests pending L2 blocks which have not yet been batch submitted, and inserts them into the tip of the local EE.
+
+These blocks are marked in the EE as not yet confirmed on L1. Replicating verifiers still run a CL, which will later mark them as L1-confirmed within the EE.
+
+### [Block Producer][block-producer]
+
+The block producer service is run by the sequencer to progress the L2 state in advance of batch submission. The block producer repeatedly calls `engine_preparePayload` to get a new L2 block and then uses `engine_executePayload` to insert the resulting block into the state. When new L1 blocks appear, it passes the corresponding deposit transactions to `engine_preparePayload`.
+
+### [Batch Submitter][batch-submitter]
+
+The batch submitter takes the pending blocks which have been inserted into the sequencer's L2 state by the block producer, and sends corresponding transactions in batches to L1.
+
+The batch submitter encodes these transactions based on the rules of [block generation][block-gen], effectively as the reverse of the process carried out by the consensus layer. As a result:
+- Blocks previously inserted by the Block Producer into the Sequencer's EE will now appear in L1 Verifiers' EEs.
+- Blocks marked as sequencer-confirmed in both the Sequencer and in Replicating verifiers will now be marked as L1-confirmed.
```

Comment:
I have a strong preference to keep them separate, I think it will be easier from a devops perspective. Imagine having to take down a rollup node due to there being a bug in the batch submission logic. Also what if we run multiple sets of rollup node + ee in clique or some other POA consensus for redundency, zero downtime rollovers etc - we will need to configure only 1 of the nodes to run with the batch submission logic. I think the spec should specify the interface the batch submitter needs to access all of the data it needs and then the implementation(s) can implement the interface

Reply:
I think we can put them in the same Go repo, and organize them as packages, and then figure out if we want to run them separately or as part of the same node later.